### PR TITLE
Feat(VoiceState): self mute/deaf methods

### PR DESCRIPTION
**Pull request by MrJacz**
**Please describe the changes this PR makes and why it should be merged:**
I've added `setSelfDeaf` and `setSelfMute` to `VoiceState` which deafens/mutes the bot on the client-side instead of server-side.

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by MrJacz** [comment]
@SpaceEEC I'm not sure I like this. I'll elaborate.
1st would selfDeaf/Mute be private or public?
2nd should setDeaf/Mute have some kind of convert to Boolean or a `|| false` in sendVoiceStateUpdate? As they could do something like `setDeaf("some stupid stuff");` and possibly break something? not sure how discord would handle that, I'd hope it ignores it but I don't think they should be able to set selfDeaf/Mute to anything.

p.s I hope I did what you asked.

**Comment by amishshah** [comment]
I honestly feel like this should come under `VoiceState` as `setSelfDeaf` or `setSelfMute` for the sake of consistency -- all methods that can change VoiceState reside there, and if it's called on a state for another user it can just throw an error.

File: src/client/voice/VoiceConnection.js [hunk 1]
```diff
@@ -125,6 +125,20 @@ class VoiceConnection extends EventEmitter {
...
+     * @private
+     */
+    this.selfDeaf = false;
+
+    /**
+     * The state of self_mute for the connection
+     * @type {boolean}
+     * @private
+     */
+    this.selfMute = false;
```
Comment (amishshah) [inline src/client/voice/VoiceConnection.js]: These properties should not exist on the VoiceConnection, they should just be getters, i.e. they should use `connection.voice.{selfDeaf,selfMute}`

Comment (MrJacz) [reply src/client/voice/VoiceConnection.js]: I haven't had time to work on this due to windows corrupting all my files and then grub deciding to not work. that is left over from the start of this pr before I started to do the changes discussed in dms

**Pull request closed without merging**